### DISCUSSION TITLE
MQTT Retain flag

### DISCUSSION THREAD

**Discussion by zoltanszalontay** [blakeblackshear/frigate#2822] [General]
Hi, is there any way to turn the Retain flag off for messages sent by Frigate?

Thx,
Z

**Comment by blakeblackshear** [comment]
There isn't. Which topics specifically?

**Reply by zoltanszalontay** [reply]
Thanks! It's the snapshot. I send a Pushover.net msg with the snapshot, but every time my Node-RED flow restarts, I get a repeated alert containing the last snapshot for hours ago for all my cameras.

BR,
Z

**Reply by blakeblackshear** [reply]
Normally you can identify messages that are retained on startup and avoid sending.

**Reply by zoltanszalontay** [reply]
Yes, that was my plan if Retain could not be disabled. I'll keep last id's in a dictionary in a persistent context.

**Reply by blakeblackshear** [reply]
I would suggest using the blueprint as a guide and building your automations off the events topic instead of using the snapshot endpoint.

**Reply by zoltanszalontay** [reply]
Thx, for the info, but I'm not using HA. I need a lot simpler solution that I can implement in NR. Also, I have been using Pushover for years. I need no more apps to receive notifications.

**Reply by blakeblackshear** [reply]
You can still use the events topic with NR too.

**Reply by zoltanszalontay** [reply]
That was my exact plan. 

What's the suggested way to link the messages of Event and Snapshot topics? I cannot see any Frigate specific IDs that could link them. 

What I can do is to: 
- Wait for a message sent to the Event topic that says it is the Last one and there is a snapshot corresponding to it (but sent to the Snapshot topic).
- Then wait for the next message sent to Snapshot topic (in a very short period of time) for the same camera and it contains the correct image?

Is this correct?

**Reply by blakeblackshear** [reply]
Don't try and coordinate across topics like that. You can use node red to pull images from the http api with the event id. https://docs.frigate.video/integrations/api#get-apieventsidsnapshotjpg

**Reply by zoltanszalontay** [reply]
Ah, cool. Many thanks! I missed this somehow.

**Reply by zoltanszalontay** [reply]
Working! Thanks again
![Screenshot 2022-02-19 at 22 53 24](https://user-images.githubusercontent.com/35921556/154820399-2cf4635b-c44f-41f9-974e-3d8e45119343.png)
!

**Comment by icsboyx** [comment]
Ciao, I just hat a similar problem, every time NodRed restart push the last snapshots, (retained ones)
There is the option inside mqtt component to not use retained messages:

[https://flows.nodered.org/node/node-red-contrib-mqtt-dynamictopic](url)
[https://github.com/rswilem/node-red-contrib-mqtt-dynamictopic](url)

![image](https://user-images.githubusercontent.com/92451526/199436348-32e6f8d6-c42e-45df-b242-29514ae0bf02.png)

Ciao
